### DISCUSSION TITLE
Mapping not working as expected

### DISCUSSION THREAD

**Discussion by powerdude** [riok/mapperly#1076] [General]
Hello all, 

the following fails to generate a mapping file and I'm not sure why.  Can anyone share any insights?

```csharp
public class Source
{
    public ICollection<SourceSub> Subs { get; init; }
}

public class SourceSub
{
    public string Name { get; init; }
}

public class Target
{

    private List<TargetSub> subs = new List<TargetSub>();

    public IReadOnlyList<TargetSub> Subs
    {
        get => subs;
        private set => subs = value.ToList();
    }
}

public class TargetSub
{
    public string Name { get; set; }
}

[Mapper(IncludedMembers = MemberVisibility.All)]
public static partial class Mapper
{
    public static partial Target MapToTarget(Source source);
}

```

**Comment by TimothyMakkison** [comment]
Hey, thanks for the bug report. 

This appears to be caused by the private field having the name `subs`. I'm guessing mapperly sees the matching private field and property and tries to map the source onto it. 
It's a little peculiar because Mapperly should never try to map `Subs` to `subs` unless `PropertyNameMappingStrategy.CaseInsensitive` is being used, I'm not sure why this is happening and will look into a fix.

Note that this isn't unique to private member mapping, it can be replicated by creating two properties, one upper case, one lower case.

For the time being, try using underscore prefixes or different names for private fields if this occurs, sorry 🙏 


Edit: Mapperly crashes on line `127` in `MembersMappingBuilderContext` as it attempts to add members to a dictionary.

Fix at #1077

**Comment by latonz** [comment]
Should be fixed in the latest release.
